gobblin-core/src/main/java/gobblin/source/extractor/partition/Partitioner.java

```diff
@@ -47,7 +49,7 @@ public Partitioner(SourceState state) {
    * @return map of partition intervals
    */
   public HashMap<Long, Long> getPartitions(long previousWatermark) {
```

Comment:
We have to consider renaming/refactoring this class later. I find the naming rather misleading. Further, it seems to be tied to the query-based extractor yet it has a generic name and namespace.